src/main/java/ui/UI.java

```diff
@@ -540,4 +546,42 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    /**
+     * Warns user if the Java runtime version is lower than HT's requirement
+     */
+    private void warnIfJavaVersionOutdated() {
+        JavaVersion requiredVersion, runtimeVersion;
+        String javaRuntimeVersionString = System.getProperty("java.runtime.version");
+
+        try {
+            requiredVersion = JavaVersion.fromString(REQUIRED_JAVA_VERSION);
+        } catch (IllegalArgumentException e) {
+            logger.error("Required Java Version string cannot be parsed. This should have been covered by test.");
+            return;
+        }
```

Comment:
Yes, this should never happen. I have a test that check this. Should I put
an assertion failure if it ever happens?
On Mar 14, 2016 9:16 AM, "Damith C. Rajapakse" <notifications@github.com>
wrote:

> In src/main/java/ui/UI.java
> <https://github.com/HubTurbo/HubTurbo/pull/1320#discussion_r55948036>:
>
> > @@ -540,4 +546,42 @@ public boolean isWindowMinimized() {
> >      public boolean isWindowFocused() {
> >          return mainStage.isFocused();
> >      }
> > +
> > +    /**
> > +     * Warns user if the Java runtime version is lower than HT's requirement
> > +     */
> > +    private void warnIfJavaVersionOutdated() {
> > +        JavaVersion requiredVersion, runtimeVersion;
> > +        String javaRuntimeVersionString = System.getProperty("java.runtime.version");
> > +
> > +        try {
> > +            requiredVersion = JavaVersion.fromString(REQUIRED_JAVA_VERSION);
> > +        } catch (IllegalArgumentException e) {
> > +            logger.error("Required Java Version string cannot be parsed. This should have been covered by test.");
> > +            return;
> > +        }
>
> This is not the user's java version. There is no way this exception happen
> right? If it did happen, we want to catch it during testing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1320/files#r55948036>.
>